### ISSUE TITLE
qobuz interface: number of favourite albums

### ISSUE THREAD

**Issue by nicolabotta** [medoc92/upmpdcli#100]
Is it possible that the number of favourite albums for the upmpdcli qobuz interface is limited to 46? I have more than 46 favourite albums. These are visible in a web browser and in the Android qobuz app. But in the upmpdcli qobuz interface only 46 favourite albums are visible. Adding a new album to a list of more than 46 favourite albums has no effect on the upmpdcli list. But marking one of the favourite albums displayed by the upmpdcli interface as non-favourite, makes one of the "hidden" favourite albums appear in the upmpdcli list!

**Comment by medoc92** [comment]
There is no explicit limitation to 46, but  I see that the code reads the data by slices of 45, so a bug is quite probable. I'll have a look at it, but for a quick fix, you can just edit qobuz/session.py and change 45 to any value of your choosing in the Favorites object methods.

**Comment by nicolabotta** [comment]
Great, thanks! Best, Nicola

**Comment by medoc92** [comment]
Actually, I don't seem to see this issue, I can get more than 46 albums. Are you sure that this is not an availability issue (upmpdcli will only list favourite albums if they are available for streaming).

Here follows how the stderr trace looks like (with upmpdcli started from a terminal, would also work with journalctl):

    [Qobuz/RawApi][[NOTICE]] uri: /favorite/getUserFavorites, params: {'type': 'albums', 'user_id': 558718, 'limit': 45, 'offset': 0}
    0$qobuz$: Favourite albums: got 45 at offset 0
    [Qobuz/RawApi][[NOTICE]] uri: /favorite/getUserFavorites, params: {'type': 'albums', 'user_id': 558718, 'limit': 45, 'offset': 45}
    0$qobuz$: Favourite albums: got 26 at offset 45


An I do get more than 46 albums listed (but not 71, probably because of availability).

**Comment by nicolabotta** [comment]
I am not sure of anything anymore! As I reported the issue, adding a new favourite would make the last one disappear. Thus, I though that there was a limit on the number of favourite albums that can be displayed. At the same time I might have had availability issues, as you suggest. I have tried again right now and everything works as one would expect. Thus I am closing the issue. Sorry for the noise!
